### ISSUE TITLE
Able to range between two dates.

### ISSUE THREAD

**Issue by confact** [crystal-lang/crystal#9327]
It would be nice to be able to range between two dates, probably only on days. Like example:
```crystal
(Time.utc(2020,3,1)..Time.utc).each do |t|
  puts t.to_s
end
```
This will give error on `succ` method not exist on `Time`
When added a struct extension:
```crystal
struct Time
  def succ
    self + 1.day
  end
end
```
The above will now work and give the following:
```bash
....
2020-05-09 00:00:00 UTC
2020-05-10 00:00:00 UTC
2020-05-11 00:00:00 UTC
2020-05-12 00:00:00 UTC
2020-05-13 00:00:00 UTC
2020-05-14 00:00:00 UTC
2020-05-15 00:00:00 UTC
2020-05-16 00:00:00 UTC
2020-05-17 00:00:00 UTC
2020-05-18 00:00:00 UTC
2020-05-19 00:00:00 UTC
2020-05-20 00:00:00 UTC
```

Would be nice to be able to range between two dates.

**Comment by jhass** [comment]
Well the problem is which unit to pick? Now you want dates, but why would hours be any less sensible?

**Comment by straight-shoota** [comment]
Time is non-discriminatory, so a `Time` instance doesn't have a natural successor and predecessor. `#succ` and `#pred` methods would just be random, so they don't exist. That's the same as for `Float` numbers for example.

But `Range` has `step` methoud which could work for this, because it allows to specify the interval between instances:
```cr
(Time.utc(2020,3,1)..Time.utc).step(1.day) do |t|
  puts t.to_s
end
```
This doesn't work currently because `Range#step` is based on calling `#succ` repeatedly on the cursor.
But it could be fixed. Maybe `Range#step` could use `#+` if it's defined.
Example: https://carc.in/#/r/93vh

This would be great with #2549 but it might be fine without it.

**Comment by jhass** [comment]
Two alternative or supplementary APIs that should cover the usecases:

1. Make  `Time` implement the `Number#to` interface (sort of), with a mandatory `step : Time::Span` interface: `Time.now.to(end_date, step: 1.day)`
2. Add `Time.every(range : Range(Time, Time), *, step : Time::Span, &)`,  so you can pass around time ranges and iterate them with `Time.every(window, step: 1.hour)`

**Comment by straight-shoota** [comment]
Number 1 looks good. I initially thought about this, too but figured `Range#step` would be more versatile. Doesn't hurt to have both, though.

Number 2. is essentially just a different syntax for my proposal to enhance `Range#step`. If that works out, I don't think there should be a separate method just for `Time`.

**Comment by confact** [comment]
@jhass I picked days here as an example and as I needed it. so that's what I had in my head. So I don't think we should only support days. Of course, we should support: year, month, day, hour, minute, seconds.

I do like the first alternative as well. It makes more sense for me to use.

**Comment by z64** [comment]
I think that `#step` would be the most idiomatic, if possible. `Time.every` gives me the impression of a method that would do timing/scheduling of a task.

**Comment by jwoertink** [comment]
I like @straight-shoota suggestion of using `step`. It would also be nice if you tried to use the range with Time, and it gave you a nice error message saying "You can't iterate over time range directly without calling the `step` method...." or something like that

**Comment by RX14** [comment]
I think we should support `Range#step` for both `+` and `succ` types.

I'm also fine with adding `Time#to` in addition to these.

**Comment by straight-shoota** [comment]
Resolved by #10279

**Comment by confact** [comment]
@straight-shoota, thanks for the great work!
